File: src/client/java/net/legitimoose/bot/discord/command/staff/Restart.java
package net.legitimoose.bot.discord.command.staff;

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

public class Restart extends ListenerAdapter {
    @Override
    public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
        if (!event.getName().equals("restart")) return;
        event.deferReply(true).queue();
        event.getHook().sendMessage("Restarting bot...").complete();
        System.exit(0);
    }
}

File: src/client/kotlin/net/legitimoose/bot/discord/staff/Restart.kt
package net.legitimoose.bot.discord.staff

import kotlin.system.exitProcess
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.legitimoose.bot.discord.Command

class Restart(override val event: SlashCommandInteractionEvent) : Command {
  override fun onCommandReceived() {
    event.deferReply(true).queue()
    event.hook.sendMessage("Restarting bot...").complete()
    exitProcess(0)
  }
}
